frontends/PyCDE/src/pycde/pycde_types.py

```diff
@@ -60,15 +60,30 @@ def declare_types(self, mod):
     if not self.registered_aliases:
       return
 
-    type_scopes = [
-        op for op in mod.body.operations if isinstance(op, hw.TypeScopeOp)
-    ]
-    if len(type_scopes) == 0:
-      with mlir.ir.InsertionPoint.at_block_begin(mod.body):
-        type_scopes.append(hw.TypeScopeOp.create(self.TYPE_SCOPE))
+    type_scopes = list()
+    for op in mod.body.operations:
```

Comment:
It's a shame there isn't a generic walk... I guess this is the best we can do for now.

Reply:
Yeah